src/org/traccar/database/CommandsManager.java

```diff
@@ -45,11 +45,15 @@ public boolean checkDeviceCommand(long deviceId, long commandId) {
         return !getAllDeviceItems(deviceId).contains(commandId);
     }
 
-    public boolean sendCommand(long commandId, long deviceId) throws Exception {
-        return sendCommand(getById(commandId), deviceId);
-    }
-
-    public boolean sendCommand(Command command, long deviceId) throws Exception {
+    public boolean sendCommand(Command command) throws Exception {
+        long deviceId = command.getDeviceId();
+        if (command.getId() != 0) {
+            Command savedCommand = getById(command.getId());
```

Comment:
Can we instead clone the object and set `deviceId`? Will be shorter and easier to manage. We won't need to modify the code in case we add something else to the command.

Reply:
There is no `clone()` by default. We should either implement interface `Cloneable` on `Command` or copying constructor.
Copying constructor also can just enumerate all setters (like here, but it will be in model and chance to miss something is less) or use reflection.

What do you prefer?

Reply:
I think clone should work just fine here.